### PR TITLE
Beautiful -v output from ansible

### PR DESCRIPTION
**Pull request by jpic**
When supporting openshift-ansible, users share pastes like (ie. a few hours ago on openshift IRC channel)::

TASK: [openshift_serviceaccounts | Grant the user access to the privileged scc] ***
changed: [li1491-86.members.linode.com] => (item=['router', {'cmd': ['oc', 'get', 'scc', 'privileged', '-o', 'yaml'], 'end': '2016-05-05 13:31:50.216857', 'stderr': u'', 'stdout': 'allowHostDirVolumePlugin: true\nallowHostIPC: true\nallowHostNetwork: true\nallowHostPID: true\nallowHostPorts: true\nallowPrivilegedContainer: true\nallowedCapabilities: null\napiVersion: v1\ndefaultAddCapabilities: null\nfsGroup:\n  type: RunAsAny\ngroups:\n- system:cluster-admins\n- system:nodes\nkind: SecurityContextConstraints\nmetadata:\n  annotations:\n    kubernetes.io/description: \'privileged allows access to all privileged and host\n      features and the ability to run as any user, any group, any fsGroup, and with\n      any SELinux context.  WARNING: this is the most relaxed SCC and should be used\n      only for cluster administration. Grant with caution.\'\n  creationTimestamp: 2016-05-05T13:30:06Z\n  name: privileged\n  resourceVersion: "371"\n  selfLink: /api/v1/securitycontextconstraints/privileged\n  uid: 7ae22005-12c5-11e6-9bc5-06174e73e52a\npriority: null\nreadOnlyRootFilesystem: false\nrequiredDropCapabilities: null\nrunAsUser:\n  type: RunAsAny\nseLinuxContext:\n  type: RunAsAny\nsupplementalGroups:\n  type: RunAsAny\nusers:\n- system:serviceaccount:openshift-infra:build-controller\n- system:serviceaccount:management-infra:management-admin\n- system:serviceaccount:management-infra:inspector-admin\nvolumes:\n- \'*\'', 'item': 'privileged', 'changed': False, 'rc': 0, 'failed': False, 'warnings': [], 'delta': '0:00:00.264340', 'invocation': {'module_name': u'command', 'module_complex_args': {}, 'module_args': u'oc get scc privileged -o yaml'}, 'stdout_lines': ['allowHostDirVolumePlugin: true', 'allowHostIPC: true', 'allowHostNetwork: true', 'allowHostPID: true', 'allowHostPorts: true', 'allowPrivilegedContainer: true', 'allowedCapabilities: null', 'apiVersion: v1', 'defaultAddCapabilities: null', 'fsGroup:', '  type: RunAsAny', 'groups:', '- system:cluster-admins', '- system:nodes', 'kind: SecurityContextConstraints', 'metadata:', '  annotations:', "    kubernetes.io/description: 'privileged allows access to all privileged and host", '      features and the ability to run as any user, any group, any fsGroup, and with', '      any SELinux context.  WARNING: this is the most relaxed SCC and should be used', "      only for cluster administration. Grant with caution.'", '  creationTimestamp: 2016-05-05T13:30:06Z', '  name: privileged', '  resourceVersion: "371"', '  selfLink: /api/v1/securitycontextconstraints/privileged', '  uid: 7ae22005-12c5-11e6-9bc5-06174e73e52a', 'priority: null', 'readOnlyRootFilesystem: false', 'requiredDropCapabilities: null', 'runAsUser:', '  type: RunAsAny', 'seLinuxContext:', '  type: RunAsAny', 'supplementalGroups:', '  type: RunAsAny', 'users:', '- system:serviceaccount:openshift-infra:build-controller', '- system:serviceaccount:management-infra:management-admin', '- system:serviceaccount:management-infra:inspector-admin', 'volumes:', "- '*'"], 'failed_when_result': False, 'start': '2016-05-05 13:31:49.952517'}])
changed: [li1491-86.members.linode.com] => (item=['registry', {'cmd': ['oc', 'get', 'scc', 'privileged', '-o', 'yaml'], 'end': '2016-05-05 13:31:50.216857', 'stderr': u'', 'stdout': 'allowHostDirVolumePlugin: true\nallowHostIPC: true\nallowHostNetwork: true\nallowHostPID: true\nallowHostPorts: true\nallowPrivilegedContainer: true\nallowedCapabilities: null\napiVersion: v1\ndefaultAddCapabilities: null\nfsGroup:\n  type: RunAsAny\ngroups:\n- system:cluster-admins\n- system:nodes\nkind: SecurityContextConstraints\nmetadata:\n  annotations:\n    kubernetes.io/description: \'privileged allows access to all privileged and host\n      features and the ability to run as any user, any group, any fsGroup, and with\n      any SELinux context.  WARNING: this is the most relaxed SCC and should be used\n      only for cluster administration. Grant with caution.\'\n  creationTimestamp: 2016-05-05T13:30:06Z\n  name: privileged\n  resourceVersion: "371"\n  selfLink: /api/v1/securitycontextconstraints/privileged\n  uid: 7ae22005-12c5-11e6-9bc5-06174e73e52a\npriority: null\nreadOnlyRootFilesystem: false\nrequiredDropCapabilities: null\nrunAsUser:\n  type: RunAsAny\nseLinuxContext:\n  type: RunAsAny\nsupplementalGroups:\n  type: RunAsAny\nusers:\n- system:serviceaccount:openshift-infra:build-controller\n- system:serviceaccount:management-infra:management-admin\n- system:serviceaccount:management-infra:inspector-admin\nvolumes:\n- \'*\'', 'item': 'privileged', 'changed': False, 'rc': 0, 'failed': False, 'warnings': [], 'delta': '0:00:00.264340', 'invocation': {'module_name': u'command', 'module_complex_args': {}, 'module_args': u'oc get scc privileged -o yaml'}, 'stdout_lines': ['allowHostDirVolumePlugin: true', 'allowHostIPC: true', 'allowHostNetwork: true', 'allowHostPID: true', 'allowHostPorts: true', 'allowPrivilegedContainer: true', 'allowedCapabilities: null', 'apiVersion: v1', 'defaultAddCapabilities: null', 'fsGroup:', '  type: RunAsAny', 'groups:', '- system:cluster-admins', '- system:nodes', 'kind: SecurityContextConstraints', 'metadata:', '  annotations:', "    kubernetes.io/description: 'privileged allows access to all privileged and host", '      features and the ability to run as any user, any group, any fsGroup, and with', '      any SELinux context.  WARNING: this is the most relaxed SCC and should be used', "      only for cluster administration. Grant with caution.'", '  creationTimestamp: 2016-05-05T13:30:06Z', '  name: privileged', '  resourceVersion: "371"', '  selfLink: /api/v1/securitycontextconstraints/privileged', '  uid: 7ae22005-12c5-11e6-9bc5-06174e73e52a', 'priority: null', 'readOnlyRootFilesystem: false', 'requiredDropCapabilities: null', 'runAsUser:', '  type: RunAsAny', 'seLinuxContext:', '  type: RunAsAny', 'supplementalGroups:', '  type: RunAsAny', 'users:', '- system:serviceaccount:openshift-infra:build-controller', '- system:serviceaccount:management-infra:management-admin', '- system:serviceaccount:management-infra:inspector-admin', 'volumes:', "- '*'"], 'failed_when_result': False, 'start': '2016-05-05 13:31:49.952517'}])

With this patch, the json is nicely indented and std{err,out} are
properly printed by default with -v.

### CODE REVIEW

**Comment by detiber** [comment]
@abutcher what are your thoughts on this approach vs the callback plugin you've been using?

It probably is about time to think about getting something in place for making the output more bearable.

**Comment by jpic** [comment]
@detiber what do you mean "in-place" ? that callback is not originally from me, it's been passed around in the ansible user community for years and has been traced in several places such as 

https://gist.github.com/cliffano/9868180
http://blog.cliffano.com/2014/04/06/human-readable-ansible-playbook-log-output-using-callback-plugin/
https://github.com/novafloss/ansible-setup/blob/master/plugins/callback/default.py

Perhaps @abutcher uses one of these ?

**Comment by abutcher** [comment]
> https://gist.github.com/cliffano/9868180
http://blog.cliffano.com/2014/04/06/human-readable-ansible-playbook-log-output-using-callback-plugin/
https://github.com/novafloss/ansible-setup/blob/master/plugins/callback/default.py
>
> I honnestly have no idea about the actual history of this, it's never been traced in any repo, just passed around as a snippet ...
>
> Perhaps @abutcher uses one of these ?

Yep I've been using a variation of human_log.py, the first one listed there, which is nice for seeing everything that is being ran and returned but is probably overkill for regular use (and obviously only works with ansible v1 as implemented there). This callback provides much nicer output when things explode which is probably the approach we'd want to take if we included one.

**Comment by jpic** [comment]
If you're into screenshots:

**Comment by jpic** [comment]
If you're into screenshots: 2016-05-06-212907_742x585_scrot

**Comment by jpic** [comment]
Do you think we should rather propose it to ansible itself ? Or rather have this within openshift-ansible ?

**Comment by detiber** [comment]
@jpic I think this is something valuable to add here (of course trying to get better output in upstream ansible would be great, but it would not be available until ansible 2.2 at the earliest).

We are currently dealing with a few major refactors (and their subsequent fallout) that is keeping us from evaluating this more fully.

**Comment by detiber** [comment]
@abutcher ptal

**Comment by tbielawa** [comment]
MMMM. That output is way more palatable.

**Comment by abutcher** [comment]
@detiber All for merging this when the 2.1 swap occurs

**Comment by abutcher** [comment]
aos-ci-test

**Comment by abutcher** [comment]
ok to test

**Comment by detiber** [comment]
pylint "errors":
```
Testing files:  callback_plugins/default.py
************* Module default
C:  1, 0: Missing module docstring (missing-docstring)
C: 19, 0: Invalid constant name "path" (invalid-name)
C: 20, 0: Invalid constant name "default_path" (invalid-name)
C: 21, 0: Invalid constant name "default" (invalid-name)
C: 24, 0: Missing class docstring (missing-docstring)
W: 24, 0: Class has no __init__ method (no-init)
C: 29, 4: Missing method docstring (missing-docstring)
E: 29, 4: Use of super on an old style class (super-on-old-class)
R: 24, 0: Too few public methods (0/2) (too-few-public-methods)
```

**Comment by jpic** [comment]
Made `$ WORKSPACE=. ./git/pylint.sh 506b993d30b46d2106d15a2ed3a46b4de3374818  add2ebacbafdf2f0fcae2991b138f71cbcfa4d43 master` pass locally.

**Comment by jpic** [comment]
It's passing now, thanks for the heads up !

**Comment by sdodson** [comment]
@jpic Can you squash those down?

**Comment by jpic** [comment]
Absolutely, sorry my laptop was in hardware failure the last 4 days ^^

File: callback_plugins/default.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+# For some reason this has to be done
+import imp
+import os
+
+ANSIBLE_PATH = imp.find_module('ansible')[1]
+DEFAULT_PATH = os.path.join(ANSIBLE_PATH, 'lugins/callback/default.py')
```
Comment (kaistian) [inline callback_plugins/default.py]: Is there a little typo, missing p in plugins?
Or maybe it should be 'callback_plugins/default.py'

Comment (jpic) [inline callback_plugins/default.py]: Nice, thanks !

**Comment by sdodson** [comment]
aos-ci-test

**Comment by jpic** [comment]
FTR we have an upstream PR: https://github.com/ansible/ansible/pull/16839

**Comment by detiber** [comment]
@abutcher @sdodson maybe we should pick up the upstream fix at some point post 3.3?

**Pull request merged**